python-threatexchange/threatexchange/cli/match_cmd.py

```diff
@@ -215,6 +220,52 @@ def execute(self, settings: CLISettings) -> None:
                             fetched_data,
                         )
 
+    def validate_hashes_signal_type(
+        self,
+        hashes_grouped_by_prefix: t.Dict[t.Optional[t.Type[SignalType]], t.Set[str]],
+        signal_types: t.List[t.Type[SignalType]],
+    ) -> None:
+        if (
+            len(hashes_grouped_by_prefix) > 2
+            and None in hashes_grouped_by_prefix.keys()
+        ):
+            raise CommandError(
+                f"Error: Provided more than one SignalType and some hashes are missing a prefix",
+                2,
```

Comment:
Consider putting what the user should fix in these cases - it's clear that something is wrong, but what the user should do to fix it should be in the message.

For this one however, you might not be adding anything by this message - it's likely captured by one of the other messages below (add a prefix)